Sources/Build/BuildOperation.swift

```diff
@@ -130,7 +130,21 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
 
         buildSystemDelegate?.buildComplete(success: success)
         delegate?.buildSystem(self, didFinishWithResult: success)
-        guard success else { throw Diagnostics.fatalError }
+        guard success else {
+            if let errorMessages = self.buildSystemDelegate?.targetErrorMessages {
+                errorMessages.forEach { key, value in
+                    value.forEach { entry in
+                        if entry.contains("could not build Objective-C module") || entry.contains("no such module") {
```

Comment:
I wonder if there is a solution that doesn't rely on parsing the error messages, which seems to me a high coupling. If any of the messages is changed (or maybe localised), then this code becomes broken, although likely the break can be caught by the test cases.

Is it possible to use type information instead of error messages? Something like this:

```swift
if let targetErrors = self.buildSystemDelegate?.targetErrors {
    targetErrors.forEach {
        ...
        switch $0 {
        case .couldNotBuildObjectiveCModule, .noSuchModule: ...
        case ...
        }
        ...
    }
}
```

This way we can keep the human languages for humans, and "machine languages" for machines.

I think another benefit of doing it like this is that any error resulted from changes to `buildSystemDelegate` can be caught at compile-time or during static analysis, instead of at run time.

Reply:
Yeah that's some good feedback. So I don't think there currently exists any type information for the errors, but that doesn't mean we can't introduce it :) 

I'll work on adding that so increase cohesion and keep coupling at a minimum